exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/QueryResultHandler.java

```diff
@@ -189,20 +218,10 @@ private UserResultsListener newUserResultsListener(QueryId queryId) {
       if ( null == resultsListener ) {
         resultsListener = bl;
       }
-      // TODO:  Is there a more direct way to detect a Query ID in whatever state this string comparison detects?
-      if ( queryId.toString().isEmpty() ) {
```

Comment:
You removed this because it's impossible to receive a batch without a queryId, right ?